tests/internal/fstore.c

```diff
@@ -78,7 +78,249 @@ void cb_all()
+    for (i = 0; i < n_files; ++i) {
+        ret = flb_fstore_file_content_copy(fs, files[i], &out_buf, &out_size);
+        TEST_CHECK(ret == 0);
+
+        TEST_CHECK(memcmp(out_buf, "fluent-bit\nfluent-bit-2\n", 11+13) == 0);
+        TEST_CHECK(out_size == 11+13);
+        flb_free(out_buf);
+    }
+    
+    /* test inactive and delete */
+    for (i = 0; i < n_files / 4; ++i) {
+        flb_fstore_file_inactive(fs, files[i]);
+    }
+
+    for (i = 0; i < n_files / 4; ++i) {
+        flb_fstore_file_delete(fs, files[i + n_files / 4]);
+    }
+
+    /* test add more files */
+    for (i = 0; i < n_files / 4 * 2; ++i) {
+        sprintf(filepath, FSF_STORE_PATH "/abc/files-round_2-%d.txt", i);
+        sprintf(filename, "files-round_2-%d.txt", i);
+
+        files[i] = flb_fstore_file_create(fs, st, filename, 100);
+        TEST_CHECK(files[i] != NULL);
+        if (!files[i]) {
+            return;
+        }
+
+        ret = stat(filepath, &st_data);
+        TEST_CHECK(ret == 0);
+
+        ret = flb_fstore_file_append(fs, files[i], "fluent-bit-3\n", 13);
+        TEST_CHECK(ret == 0);
+
+        ret = flb_fstore_file_content_copy(fs, files[i], &out_buf, &out_size);
+        TEST_CHECK(ret == 0);
+
+        TEST_CHECK(memcmp(out_buf, "fluent-bit-3\n", 13) == 0);
+        TEST_CHECK(out_size == 13);
+        flb_free(out_buf);
+    }
+
+    /* copy, delete, inactivate, and create */
+    for (i = 0; i < n_files; ++i) {
+        ret = flb_fstore_file_content_copy(fs, files[i], &out_buf, &out_size);
+        TEST_CHECK(ret == 0);
+        if (i < n_files / 4 * 2) {
+            TEST_CHECK(memcmp(out_buf, "fluent-bit-3\n", 13) == 0);
+            TEST_CHECK(out_size == 13);
+            flb_free(out_buf);
+        }
+        else {
+            TEST_CHECK(memcmp(out_buf, "fluent-bit\nfluent-bit-2\n", 11+13) == 0);
+            TEST_CHECK(out_size == 11+13);
+            flb_free(out_buf);
+        }
+       
+        if (i % 2) {
+            flb_fstore_file_delete(fs, files[i]);
+        } else {
+            flb_fstore_file_inactive(fs, files[i]);
+        }
+
+        sprintf(filepath, FSF_STORE_PATH "/abc/files-round_3-%d.txt", i);
+        sprintf(filename, "files-round_3-%d.txt", i);
+
+        files[i] = flb_fstore_file_create(fs, st, filename, 100);
+    }
+
+    flb_fstore_destroy(fs);
+}
+
+void test_add_1_file()
+{
+    test_add_n_files(1);
+}
+
+void test_add_10_files()
+{
+    test_add_n_files(10);
+}
+
+void test_add_64_files()
+{
+    test_add_n_files(64);
+}
+
+void test_add_65_files()
+{
+    test_add_n_files(65);
+}
+
+void test_add_256_files()
+{
+    test_add_n_files(256);
+}
+
+/* Logging callback, once called it just turn on the log_check flag */
+static int log_cb(struct cio_ctx *ctx, int level, const char *file, int line,
+                  char *str)
+{
+    (void) ctx;
+
+    printf("[cio-test-fs] %-60s => %s:%i\n",  str, file, line);
+    return 0;
+}
+
+void test_chunkio_up_down_up_append()
```

Comment:
This test case displays the issue with chunk io append after up > down > up > append.